### PR TITLE
Kibana spaces data source

### PR DESCRIPTION
**Pull request by jdesnoes**
Adding a data source to search Kibana spaces by their name.

### CODE REVIEW

**Comment by tobio** [comment]
@jdesnoes there's a strong preference for all new resources/data sources to be written with the [Terraform plugin framework](https://developer.hashicorp.com/terraform/plugin/framework). This would be the first data source, but the [data view](https://github.com/elastic/terraform-provider-elasticstack/tree/main/internal/kibana/data_view) resource might be a good example to work from.

**Comment by tobio** [comment]
The lint failure is due to automatic docs generation. Are you able to add a [template](https://github.com/elastic/terraform-provider-elasticstack/tree/main/templates/data-sources) for this data source, along with an [example](https://github.com/elastic/terraform-provider-elasticstack/tree/main/examples/data-sources) and then run `make docs-generate` and commit everything.

**Comment by tobio** [comment]
FYI you can run the acceptance tests locally against a docker environment with [`make docker-testacc`](https://github.com/elastic/terraform-provider-elasticstack/blob/main/Makefile#L74), or your test specifically with `make docker-testacc TESTARGS='-run ^TestAccDataSourceKibanaSpaces$$'`

**Comment by jdesnoes** [comment]
Thanks for all your advices @tobio, I just committed a new version of the Kibana spaces data source based on the Terraform plugin framework, with acceptances tests working and documentation generated.

File: internal/kibana/spaces/data_source.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/disaster37/go-kibana-rest/v8/kbapi"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &dataSource{}
+	_ datasource.DataSource              = &dataSource{}
```
Comment (tobio) [inline internal/kibana/spaces/data_source.go]:
```suggestion
	_ datasource.DataSource              = &dataSource{}
```

File: internal/kibana/spaces/read.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			Color:       types.StringValue(space.Color),
+			ImageUrl:    types.StringValue(space.ImageURL),
+		}
+
+		if disabledFeatures, diags := types.ListValueFrom(ctx, types.StringType, space.DisabledFeatures); diags.HasError() {
+			resp.Diagnostics.Append(diags...)
+			return
+		} else {
+			spaceState.DisabledFeatures = disabledFeatures
+		}
```
Comment (tobio) [inline internal/kibana/spaces/read.go]:
```suggestion
		} 
		spaceState.DisabledFeatures = disabledFeatures
		
```

**Review by tobio — DISMISSED** [review]
LGTM. There's a couple of minor changes, are you able to add an entry to the [CHANGELOG](https://github.com/elastic/terraform-provider-elasticstack/blob/main/CHANGELOG.md) for this PR as well?

**Comment by jdesnoes** [comment]
> LGTM. There's a couple of minor changes, are you able to add an entry to the [CHANGELOG](https://github.com/elastic/terraform-provider-elasticstack/blob/main/CHANGELOG.md) for this PR as well?

Thanks, it's done.

**Review by tobio — APPROVED** [review]
Thanks for adding this one!

**Pull request merged**